#!/usr/bin/python

import argparse
import sys

from yarsync import YARsync


if __name__ == "__main__":
    # parse arguments
    try:
        ys = YARsync(sys.argv)
    except (argparse.ArgumentError, argparse.ArgumentTypeError, SystemExit) \
           as err:
        # the error is printed by argparse.
        # SystemExit can raise from argparse
        # and it has caused much misunderstanding, e.g.
        # https://stackoverflow.com/questions/42249982/systemexit-2-error-when-calling-parse-args
        sys.exit(7)
    except OSError as err:
        # the error is printed by YARsync
        sys.exit(8)

    # make actual call
    # this should throw no exceptions
    returncode = ys()
    sys.exit(returncode)
